### ISSUE TITLE
EPath improvement for setting value of sub-expressions

### ISSUE THREAD

**Issue by olcc** [sympy/sympy#23228]
Hello,
I find it a bit weird how to change a sub-expression inside an expression. The first line below is impossible. The second line works, but is not very intuitive and is unnecessarily complicated.
```
EXPR.args[0].args[1] = NEW_SUB_EXPRESSION   # Impossible
epath("/[1]/[0]", EXPR, lambda subexpr: NEW_SUB_EXPRESSION)   # Works
```
Could the situation be improved in such a way that the two lines below would work?
```
EPath("/[1]/[0]").set(EXPR, NEW_SUB_EXPRESSION)
epath("/[1]/[0]", EXPR, repl=NEW_SUB_EXPRESSION)
```
This would imply creating a set() method in EPath, which would simply call apply() in the background. If you are not affraid, I could write the change and send a pull request.

**Comment by smichr** [comment]
Perhaps it could just be called `subs`. And couldn't the case of two expressions be recognized automatically as a replacement request? `replace` is already a multi-interpretation method. Does the case of `epath(path, expr1, expr2)` have any meaning, presently?

**Comment by olcc** [comment]
I disagree for the name `subs` because it implies the idea of a recursive search for the  substitution. My proposal is a replacement of the expression targeted by the `EPath`. There is no recursive search like `subs` in this case.

I also disagree with the syntax `epath(path, expr1, expr2)` because overloading can make the code more confusing. Keyword arguments leave it more clean and maintainable.
